wit/types.wit

```diff
@@ -1,41 +1,47 @@
-/// An interface grouping all necessary types for messaging.
 default interface messaging-types {
-    /// A broker type that allows the exchange of messages.
-    type broker = u32
-    drop-broker: func(b: broker)
-    open-broker: func(name: string) -> result<broker, error>
+    /// An object that allows connecting to a service (e.g., a broker).
+    type client = u32
+    drop-client: func(c: client)
+    open-client: func(name: string) -> result<client, error>
 
-    /// An event type that follows the CloudEvents specification (https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md). We
-    /// assume the type of the data is a byte sequence. It is up to the data schema to determine what type of the data payload the event 
-    /// contains.
-    record event {
-        specversion: string,
-        ty: string,
-        source: string,
-        id: string,
-        data: option<list<u8>>,
-        datacontenttype: option<string>,
-        dataschema: option<string>,
-        subject: option<string>,
-        time: option<string>,	
-        extensions: option<list<tuple<string, string>>>
-    }
-    
-    /// Channels specify where a published message should land. There are two types of channels:
-    /// - queue: competitive consumers, and
-    /// - topic: non-competitive consumers.
-    variant channel {
-        queue(string),
-        topic(string)
+    /// An entity may be a queue or a topic and is identifiable by name.
+    type entity = string
+
+    /// A message contains binary data, a format specification, and a metadata decoration of key/value pairs.
+    ///     - The format could be: "CloudEvents/1.0", "HTTP/1.1" / "HTTP/2" / "HTTP/3", "AMQP/1.0", "MQTT/3.1.1" / "MQTT/5.0", "Kafka/0.11"
+    ///     as defined per: https://github.com/clemensv/spec/blob/registry-extensions/registry/spec.md#message-formats
+    record message {
+        data: list<u8>,
```

Comment:
Since `data` needs to be qualified for any consumer to understand it, `content-type` must be special-cased as a property right on this interface. You need to be able to decode `data` without digging into the property bag. We made the same call for CloudEvents.